salesagility/SuiteCRM
# Calendar slot duration fixed

Calendar slot duration not changing

## Description
As per issue #2934 Calendar slot was not updating due to hardcoded values being set in the tpl file. These have been removed and replaced by a variable value which is set within the config/config_override values. 

Also as per the second bug on this issue, the parameters being set were out of date so these were changed to match expected parameters. This involved adding in extra functionality to format the time_slot value to the new expected value. Formatting to this new value allows for the configuration file value to remain consistent with previous versions. 

Finally, calls_per_day was removed as it looks like this is redundant now (there can likely be further refactoring to remove calculate_day_range() also).

## How To Test This
Testing functionality of the calendar by:

- Making configuration changes to day_timestep and week_timestep.
- Ensuring day, week, month, shared month, shared week slots appear consistently with set configuration.
- Ensuring it is possible to create, edit activity on each view and that the view is updated as expected.
- Ensuring my calendar dashlet slots match configuration for each view and show calendar activity as expected.

## Types of changes


### Final checklist
